test/html/scrollToElement.html

```diff
@@ -0,0 +1,155 @@
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    <p id="third">
+      <p id="third-clue"></p>
+      <a id="third-a" href="#" onclick="scrollFunction('fourth');return false;">Click me to scroll to fourth element</a>
+    </p>
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    <input id="fourth" type="text">
+    <p id="fourth-clue"></p>
+    <a id="fourth-a" href="#" onclick="scrollFunction('first');return false;">Click me to scroll to first element</a>
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+
+    <script>
+      function scrollFunction(id) {
+        let elmnt = document.getElementById(id);
+        let elmnt_a_tag = document.getElementById(id + "-a").focus();
+        let clue = document.getElementById(id + "-clue");
+        elmnt.scrollIntoView();
+      }
+
+      function elementsInViewPort() {
+        let first = document.querySelector("#first");
```

Comment:
Same as above for all these - they can be `const`.  Also, define clue up here once:

```javascript
let clue;
```

And then further down in the code, you can just keep reassigning it without the `let` keyword:

```javascript
clue = document.querySelector("#first-clue");
```